sdl_ios/SmartDeviceLink/SDLRPCNotification.m

```diff
@@ -1,9 +1,28 @@
 //  SDLRPCNotification.m
 //
-//  Copyright (c) 2014 Ford Motor Company. All rights reserved.
+//  
 
 #import <SmartDeviceLink/SDLRPCNotification.h>
+#import "SDLNames.h"
 
 @implementation SDLRPCNotification
 
+- (id)initWithName:(NSString *)name
+{
+    if (self = [super initWithName:name])
+    {
+        messageType = NAMES_notification;
+    }
+    return self;
+}
+
+- (id)initWithDictionary:(NSMutableDictionary*) dict
+{
+    if (self = [super initWithDictionary:dict])
+    {
+        messageType = NAMES_notification;
+    }
+    return self;
+}
```

Comment:
Some things to note about this fix:
1. The incorrectly set `messageType` issue does not just affect Notifications, but Responses as well
2. Unfortunately, what you currently have will not quite fix the issue. The reason for this is that while the `messageType` variable **will** be set correctly, this value is used in creating the internal dictionary, and thus the dictionary will be improperly structured.
```objc
-(id) initWithName:(NSString*) name {
	if (self = [super init]) {
		function = [[NSMutableDictionary alloc] initWithCapacity:3];
		parameters = [[NSMutableDictionary alloc] init];
		messageType = NAMES_request;
		[store setObject:function forKey:messageType];
		[function setObject:parameters forKey:NAMES_parameters];
		[function setObject:name forKey:NAMES_operation_name];
	}
	return self;
}
```
As you can see, the `messageType` is used as a key in the dictionary, and this key-value pair is set while the `messageType` is still "request".
```objc
messageType = NAMES_request;
[store setObject:function forKey:messageType];
```
The simplest way to fix this issue would be to remove these two lines from [SDLRPCMessage initWithDictionary] and add them to the override of initWithDictionary in SDLRPCNotification, SDLRPCResponse, SDLRPCRequest, replacing `names_request` with `names_notification`, `names_response`, and `names_request` respectively.